rfcs/0007-filtering.md

```diff
@@ -0,0 +1,127 @@
+* [Proposed Solution](#proposed-solution)
+    * [Proposed Solution: Detailed Design](#proposed-solution-detailed-design)
+    * [Proposed Solution: Unresolved Questions](#proposed-solution-unresolved-questions)
+    * [Proposed Solution: Drawbacks](#proposed-solution-drawbacks)
+    * [Proposed Solution: Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Future Possibilities](#future-possibilities)
+
+## Motivation
+[Motivation]: #motivation
+
+AB2D receives lots of data back from BFD when we make API calls to gather EOBs (Explanation of Benefits), 
+where we filter out much of the data we receive. This might not be terrible for a few API calls, but for 
+thousands it can impact our performance significantly. Since we only need a small portion of the data that we receive, it makes
+sense for the server to send us only the data that we end up using. This will also help BFD in reducing
+the amount of data that they need to send over the network.  
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+The solution will allow BFD to keep track of which fields AB2D will use on the server side so that any change in the fields 
+that AB2D receives would need to be updated in the BFD code base. Since we don't expect these fields to change too often, 
+we can have them stored in the BFD code base. The filtering will need to happen towards the end of processing. This is to avoid
+spaghetti code since transformations from the entity object to the FHIR object happen in several places throughout
+the codebase. 
+
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+The BeanUtils library can be utilized for transforming properties
+
+```
+<dependency>
+    <groupId>commons-beanutils</groupId>
+    <artifactId>commons-beanutils</artifactId>
+    <version>1.8.2</version>
+</dependency>
+```
+
+A `private static final Set<String>` of fields will need to be maintained in `TransformerUtils`. This will
+be an allow list of fields that AB2D wants to keep in the FHIR object that gets sent back to the client. Just before the
+end of the method `findByPatient` in `ExplanationOfBenefitsResourceProvider` the `eobs` can be passed to a new
+method in `TransformerUtils`.
+
+This method will be doing the bulk of the work that we are interested in for this RFC. The filtering will happen
+in this method where each field in the EOBs are visited. Reflection will be used to gain the list of all 
+fields in the object we are interested in (ExplanationOfBenefits). If the field matches a name in the allowed list it will 
+be allowed to stay, otherwise it will be nulled out. Every field in the object will need to be visited so that we don't miss any
+fields.
+
+```
+private static final Set<String> allowAB2DList = new HashSet<>();
+
+static {
+    allowedAB2DList.add("patient");
+    allowedAB2DList.add("provider");
+    allowedAB2DList.add("organization");
+    ...
+}
```

Comment:
How will this account for nested fields? For example, `item` is a very large bucket of other fields, and I imagine you need to filter out many of those, also.